src/client/texture_pool.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+Minetest
+Copyright (C) 2019 kilbith, Jean-Patrick Guerrero <jeanpatrick.guerrero@gmail.com>
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "texture_pool.h"
+#include "porting.h"
+
+s32 TexturePool::addTexture(const std::string &name,
+		const std::string &base_name, s32 frame_count,
+		u64 frame_duration)
+{
+	s32 texture_idx = textures.size() + 1;
+	textures.resize(texture_idx);
+	textures[texture_idx - 1].set(name, base_name, frame_count, frame_duration);
+	animations.push_back(texture_idx - 1);
+
+	return texture_idx;
+}
+
+s32 TexturePool::addTexture(const std::string &name)
+{
+	s32 texture_idx = textures.size() + 1;
+	textures.resize(texture_idx);
+	textures[texture_idx - 1].set(name);
```

Comment:
instead of resize + set => push_back

Reply:
This is not `std::set` but `Texture::set`